x-pack/plugins/fleet/server/routes/output/handler.ts

```diff
@@ -166,21 +166,31 @@ async function validateOutputServerless(
   if (!cloudSetup?.isServerlessEnabled) {
     return;
   }
+
   if (output.type === outputType.RemoteElasticsearch) {
     throw Boom.badRequest('Output type remote_elasticsearch not supported in serverless');
   }
+
   // Elasticsearch outputs must have the default host URL in serverless.
-  // No need to validate on update if hosts are not passed.
-  if (outputId && !output.hosts) {
+  // No need to validate for other types.
+  if (output.type !== outputType.Elasticsearch) {
```

Comment:
I think the type here could be undefined as our `PUT` are in fact `PATCH`

Reply:
Good catch, thanks. I've modified the unit tests slightly to test that update works as expected when the output type is not passed.